Python/bytecodes.c

```diff
@@ -131,10 +132,42 @@ dummy_func(
         inst(NOP, (--)) {
         }
 
+        inst(INSTRUMENTED_RESUME, (--)) {
+            /* Check monitoring *before* calling instrument */
+            /* Possibly combine this with eval breaker */
+            if (frame->f_code->_co_instrumentation_version != tstate->interp->monitoring_version) {
+                if (_Py_Instrument(frame->f_code, tstate->interp)) {
+                    goto error;
+                }
+                next_instr--;
+            }
+            else {
+                _PyFrame_SetStackPointer(frame, stack_pointer);
+                int err = _Py_call_instrumentation(
+                        tstate, oparg > 0, frame, next_instr-1);
+                stack_pointer = _PyFrame_GetStackPointer(frame);
+                ERROR_IF(err, error);
+                if (frame->prev_instr != next_instr-1) {
+                    /* Instrumentation has jumped */
+                    next_instr = frame->prev_instr;
+                    DISPATCH();
+                }
+                if (_Py_atomic_load_relaxed_int32(eval_breaker) && oparg < 2) {
+                    goto handle_eval_breaker;
+                }
+            }
+        }
+
         inst(RESUME, (--)) {
```

Comment:
Not to be too picky, but could you be consistent about whether the INSTRUMENTED version of an opcode comes before or after the original opcode? Most of the time INSTRUMENTED comes second.